src/mimeparser.rs

```diff
@@ -745,15 +763,39 @@ impl MimeMessage {
             .and_then(|msgid| parse_message_id(msgid).ok())
     }
 
-    fn merge_headers(headers: &mut HashMap<String, String>, fields: &[mailparse::MailHeader<'_>]) {
+    fn merge_headers(
+        context: &Context,
+        headers: &mut HashMap<String, String>,
+        recipients: &mut MailAddrList,
+        from: &mut MailAddrList,
+        fields: &[mailparse::MailHeader<'_>],
+    ) {
         for field in fields {
             // lowercasing all headers is technically not correct, but makes things work better
             let key = field.get_key().to_lowercase();
             if !headers.contains_key(&key) || // key already exists, only overwrite known types (protected headers)
                     is_known(&key) || key.starts_with("chat-")
             {
-                let value = field.get_value();
-                headers.insert(key.to_string(), value);
+                if key == "cc" || key == "to" {
+                    match mailparse::addrparse_header(field) {
+                        Ok(mut addrlist) => {
+                            recipients.clear();
```

Comment:
What happens if there is both a `To` and `Cc` header? When `Cc` is parsed, this line will drop every recipient from the `To` field.

Reply:
Right, fixed this.